azure-pipelines/run_test_steps.yml

```
@@ -28,7 +28,9 @@ parameters:
 steps:
 - checkout: self
   clean: true
-  fetchDepth: 1
+  # A minimum dept of 2 is required, as for PR head is the auto-merge commit
+  # and we need the last commit from PR to report the coverage.
```

This is very possibly a topic for a separate PR, but what about just skipping the automatic merge commit and [checking out the actual branch](https://github.com/actions/checkout#Checkout-pull-request-HEAD-commit-instead-of-merge-commit) instead?  It's not like the automatic merge commit actually keeps the CI report up to date with the target branch.  Relatedly, having dropped the requirement that branches are up to date seems a bit hazardous.

```suggestion
  # A minimum depth of 2 is required, as the default is the auto-merge commit
  # and we need the last commit from the PR to report the coverage.
```

What code actually looks back up to the parent to use this?  Maybe this relates to the long-standing complaints about sometimes getting wrong delta coverage reports?

---

Thanks for the typo fixes.

Note that this is an issue with Azure Pipelines and not GitHub actions ... so we need different docs  and I don't think there is this option

https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema%2Cparameter-schema#checkout


The codecov bash script in Azure needs this.

Running the test on branch HEAD vs  merge-commit should be debated outside of this PR.

---

Sorry, too much GHA on my mind.  Thanks for the correction.  I would expect that AP has some variable that could be used to specify the branch instead of the merge commit.  But agreed, no need for a debate here.